snapshot/snapshotfs/upload.go

```diff
@@ -590,18 +601,20 @@ func uploadDirInternal(
 	u.Progress.StartedDirectory(dirRelativePath)
 	defer u.Progress.FinishedDirectory(dirRelativePath)
 
-	var summ fs.DirectorySummary
-	summ.TotalDirCount = 1
+	dirManifest := &snapshot.DirManifest{
+		StreamType: directoryStreamType,
+		Summary: &fs.DirectorySummary{
+			TotalDirCount: 1,
+		},
+	}
 
 	defer func() {
-		summ.IncompleteReason = u.cancelReason()
+		dirManifest.Summary.IncompleteReason = u.cancelReason()
 	}()
 
-	log(ctx).Debugf("reading directory %v", dirRelativePath)
-
+	t0 := u.repo.Time()
```

Comment:
if `t0` is used for reporting elapsed time, then it is OK and maybe preferable to use `time.Now()`